src/perf_counters.h

```diff
@@ -136,17 +136,18 @@ class PerfCountersMeasurement final {
     // Tell the compiler to not move instructions above/below where we take
     // the snapshot.
     ClobberMemory();
-    counters_.Snapshot(&start_values_);
+    valid_read_ &= counters_.Snapshot(&start_values_);
     ClobberMemory();
   }
 
-  BENCHMARK_ALWAYS_INLINE std::vector<std::pair<std::string, double>>
+  BENCHMARK_ALWAYS_INLINE
+  std::pair<std::vector<std::pair<std::string, double>>, bool>
```

Comment:
i think this return type is getting to the point where we should consider a separate data structure.  i'd also be ok with a boolean return for validity and an out param for the vector of measurements.

Reply:
Add PerfCountersMeasurement::Result as the returned data structure.